dataservice/api/phenotype/resources.py

```diff
@@ -0,0 +1,162 @@
+
+
+class PhenotypeListAPI(CRUDView):
+    """
+    Phenotype REST API
+    """
+    endpoint = 'phenotypes_list'
+    rule = '/phenotypes'
+    schemas = {'Phenotype': PhenotypeSchema}
+
+    @paginated
+    def get(self, after, limit):
+        """
+        Get all phenotypes
+        ---
+        description: Get all phenotypes
+        template:
+          path:
+            get_list.yml
+          properties:
+            resource:
+              Phenotype
+        """
+        q = Phenotype.query
+
+        return (PhenotypeSchema(many=True)
+                .jsonify(Pagination(q, after, limit)))
+
+    def post(self):
+        """
+        Create a new phenotype
+        ---
+        template:
+          path:
+            new_resource.yml
+          properties:
+            resource:
+              Phenotype
+        """
+
+        body = request.json
+
+        # Deserialize
+        try:
+            p = PhenotypeSchema(strict=True).load(body).data
+        # Request body not valid
+        except ValidationError as e:
+            abort(400, 'could not create phenotype: {}'.format(e.messages))
+
+        # Add to and save in database
+        db.session.add(p)
+        db.session.commit()
+
+        return PhenotypeSchema(201, 'phenotype {} created'
+                               .format(p.kf_id)).jsonify(p), 201
+
+
+class PhenotypeAPI(CRUDView):
+    """
+    Phenotype REST API
+    """
+    endpoint = 'phenotypes'
+    rule = '/phenotypes/<string:kf_id>'
+    schemas = {'Phenotype': PhenotypeSchema}
+
+    def get(self, kf_id):
+        """
+        Get a phenotype by id
+        ---
+        template:
+          path:
+            get_by_id.yml
+          properties:
+            resource:
+              Phenotype
+        """
+        # Get one
+        try:
+            p = Phenotype.query.filter_by(kf_id=kf_id).one()
+        # Not found in database
+        except NoResultFound:
+            abort(404, 'could not find {} `{}`'
+                  .format('phenotype', kf_id))
+        return PhenotypeSchema().jsonify(p)
+
+    def patch(self, kf_id):
+        """
+        Partially update existing phenotype
+
+        Update an existing phenotype given a Kids First id
+        ---
+        template:
+          path:
+            update_by_id.yml
+          properties:
+            resource:
+              Phenotype
+        """
+        body = request.json
+        try:
+            # Check if phenotype exists
+            p1 = Phenotype.query.filter_by(kf_id=kf_id).one()
+        # Not found in database
+        except NoResultFound:
+            abort(404, 'could not find {} `{}`'.format('phenotype', kf_id))
+
+        # Validation only
+        try:
+            p = PhenotypeSchema(strict=True).load(body).data
+        # Request body not valid
+        except ValidationError as e:
+            abort(400, 'could not update phenotype: {}'.format(e.messages))
+
+        # Deserialize
```

Comment:
I think you want to remove this block here. Looks like this is left over from PUT implementation since this block of code will overwrite all of the properties on phenotype.